python/ray/data/_internal/execution/operators/union_operator.py

```diff
@@ -0,0 +1,93 @@
+from typing import List, Optional
+
+from ray.data._internal.execution.interfaces import (
+    ExecutionOptions,
+    PhysicalOperator,
+    RefBundle,
+)
+from ray.data._internal.execution.operators.base_physical_operator import NAryOperator
+from ray.data._internal.stats import StatsDict
+
+
+class UnionOperator(NAryOperator):
+    """An operator that combines blocks from input operators."""
+
+    def __init__(
+        self,
+        *input_ops: PhysicalOperator,
+    ):
+        """Create a UnionOperator.
+
+        Args:
+            input_ops: Operators generating input data for this operator to union.
+        """
+        # Intermediary buffers used to store blocks from each input dependency
```

Comment:
Nit: initialize `self. _preserve_order` to false here. 
And for the following 2 variables, we can comment "Only used when \`self._preserve_order\` is True", rather than natural language. It'd be easier to find them when searching key words.